src/main/java/io/supertokens/webserver/api/oauth/OAuthClientsAPI.java

```diff
@@ -137,11 +139,46 @@ protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws
         }
     }
 
-    private JsonObject createJsonFromException(OAuthException exception){
+    @Override
+    protected void doPatch(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+        InputParser.collectAllMissingRequiredFieldsOrThrowError(input, REQUIRED_INPUT_FIELDS_FOR_PATCH);
+
+        try {
+            AppIdentifier appIdentifier = getAppIdentifier(req);
+            Storage storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+
+            JsonObject response = OAuth.updateOauthClient(super.main, appIdentifier, storage, input);
+            JsonObject postResponseBody = new JsonObject();
+            postResponseBody.addProperty("status", "OK");
+            postResponseBody.add("client", response);
+            sendJsonResponse(200, postResponseBody, resp);
+
+        } catch (OAuthAPIInvalidInputException exception) {
+            JsonObject errorResponse = createJsonFromException(exception, INVALID_INPUT_ERROR);
+            sendJsonResponse(400, errorResponse, resp);
```

Comment:
we use the BadRequestException for this and the string response should contain the details. No need for status and stuff here